### ISSUE TITLE
Unable to deploy syndesis on openshift 

### ISSUE THREAD

**Issue by svchinche** [syndesisio/syndesis#9664]
## We were installing syndesis on openshift cluster but Operator installation is failing due to 

OC Version
------------
Client Version: 4.7.0-0.okd-2021-02-25-144700
Server Version: 4.7.0-0.okd-2021-02-25-144700

Syndesis Version
-----------------
1.11.0

Steps for installing syndesis
oc login -u <username> -p <password>
wget https://github.com/syndesisio/syndesis/releases/download/1.11.0/syndesis-cli.zip
./syndesis install -s
./syndesis install --setup --grant kubeadmin
./syndesis install -p syndesis --route <route> --tag 1.11.0

[ec2-user@ip-10-0-66-63 ~]$ oc get po
NAME                         READY   STATUS      RESTARTS   AGE
syndesis-operator-1-deploy   0/1     Completed   0          45m
syndesis-operator-1-m5xbd    0/1     Error       13         44m
[ec2-user@ip-10-0-66-63 ~]$ oc logs syndesis-operator-1-m5xbd
{"level":"info","ts":1625070183.6749618,"logger":"cmd","msg":"Go Version: go1.13.7"}
{"level":"info","ts":1625070183.6749954,"logger":"cmd","msg":"Go OS/Arch: linux/amd64"}
{"level":"info","ts":1625070183.6750052,"logger":"cmd","msg":"Version of operator-sdk: v0.17.0+git"}
{"level":"info","ts":1625070183.6750135,"logger":"cmd","msg":"Syndesis Operator Version: 1.11.0"}
{"level":"info","ts":1625070183.6750174,"logger":"cmd","msg":"Syndesis Operator Image: syndesis/syndesis-operator"}
{"level":"info","ts":1625070183.675027,"logger":"cmd","msg":"Syndesis Build Time: 2021-03-05T17:23:46-0500"}
I0630 16:23:04.727232       1 request.go:621] Throttling request took 1.036640397s, request: GET:https://172.30.0.1:443/apis/scheduling.k8s.io/v1?timeout=32s
{"level":"info","ts":1625070189.1417074,"logger":"leader","msg":"Trying to become the leader."}
{"level":"info","ts":1625070191.869927,"logger":"leader","msg":"Found existing lock with my name. I was likely restarted."}
{"level":"info","ts":1625070191.8699603,"logger":"leader","msg":"Continuing as the leader."}
{"level":"info","ts":1625070194.5849257,"logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":"0.0.0.0:8383"}
{"level":"info","ts":1625070194.5852263,"logger":"cmd","msg":"registering resource schemes."}
I0630 16:23:15.636616       1 request.go:621] Throttling request took 1.044220284s, request: GET:https://172.30.0.1:443/apis/apiextensions.k8s.io/v1?timeout=32s
error: no matches for kind "Syndesis" in version "syndesis.io/v1beta1"

**Comment by zregvart** [comment]
I had no issues installing Syndesis 1.11.0 on a fresh OpenShift cluster using these commands:

```
$ oc version
Client Version: 4.7.13
Server Version: 4.7.13
Kubernetes Version: v1.20.0+df9c838
$ oc login -u kubeadmin
$ curl -OL https://github.com/syndesisio/syndesis/releases/download/1.11.0/syndesis-cli.zip
$ unzip syndesis-cli.zip
Archive:  syndesis-cli.zip
   creating: commands/
   creating: commands/binaries/
  inflating: commands/binaries/README.md  
  inflating: commands/build          
  inflating: commands/completion     
  inflating: commands/crc            
  inflating: commands/dev            
  inflating: commands/install        
  inflating: commands/integration-test  
  inflating: commands/minishift      
  inflating: commands/release        
  inflating: commands/system-test    
   creating: commands/tests/
  inflating: commands/tests/test-oc-version.sh  
  inflating: commands/ui             
   creating: commands/util/
  inflating: commands/util/common_funcs  
  inflating: commands/util/maven_funcs  
  inflating: commands/util/openshift_funcs  
  inflating: commands/util/operator_funcs  
  inflating: syndesis   
$ ./syndesis install --setup --grant developer --cluster --tag 1.11.0
operator binary not found under /home/zregvart/.syndesis/bin/syndesis-operator, attempting to download...
operator binary successfully downloaded
Installing Syndesis CRD
I0701 11:04:53.550244   54070 request.go:621] Throttling request took 1.011026013s, request: GET:https://api.crc.testing:6443/apis/authentication.k8s.io/v1?timeout=32s
Grant permission to create Syndesis to user developer
I0701 11:05:01.379527   54114 request.go:621] Throttling request took 1.031541992s, request: GET:https://api.crc.testing:6443/apis/machineconfiguration.openshift.io/v1?timeout=32s
$ ./syndesis install --project syndesis
Creating project syndesis
Already on project "syndesis" on server "https://api.crc.testing:6443".
Deploying Syndesis operator
I0701 11:07:47.498021   55289 request.go:621] Throttling request took 1.033337354s, request: GET:https://api.crc.testing:6443/apis/apiregistration.k8s.io/v1?timeout=32s
Waiting for syndesis-operator to be scaled to 1
Sleeping 10s ...
NAME                         READY   STATUS    RESTARTS   AGE
syndesis-operator-1-deploy   0/1     Pending   0          0s
syndesis-operator-1-deploy   0/1     Pending   0          0s
syndesis-operator-1-deploy   0/1     ContainerCreating   0          0s
syndesis-operator-1-deploy   0/1     ContainerCreating   0          2s
Sleeping 10s ...
syndesis-operator-1-deploy   1/1     Running             0          10s
syndesis-operator-1-spthx    0/1     Pending             0          0s
syndesis-operator-1-spthx    0/1     Pending             0          0s
syndesis-operator-1-spthx    0/1     Init:0/1            0          0s
syndesis-operator-1-spthx    0/1     Init:0/1            0          2s
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
syndesis-operator-1-spthx    0/1     PodInitializing     0          33s
Sleeping 10s ...
Sleeping 10s ...
syndesis-operator-1-spthx    1/1     Running             0          55s
syndesis-operator-1-deploy   0/1     Completed           0          67s
Deploying syndesis app.
I0701 11:09:12.434955   56742 request.go:621] Throttling request took 1.015901368s, request: GET:https://api.crc.testing:6443/apis/flowcontrol.apiserver.k8s.io/v1alpha1?timeout=32s
Install finished.
$ oc get --namespace syndesis pods
NAME                           READY   STATUS              RESTARTS   AGE
syndesis-db-1-deploy           1/1     Running             0          64s
syndesis-db-1-rblqb            0/2     ContainerCreating   0          52s
syndesis-meta-1-5cnwj          0/1     ContainerCreating   0          60s
syndesis-meta-1-deploy         1/1     Running             0          68s
syndesis-oauthproxy-1-4f6ct    0/1     ContainerCreating   0          58s
syndesis-oauthproxy-1-deploy   1/1     Running             0          67s
syndesis-operator-1-deploy     0/1     Completed           0          2m42s
syndesis-operator-1-spthx      1/1     Running             0          2m31s
syndesis-prometheus-1-deploy   1/1     Running             0          65s
syndesis-prometheus-1-wd6g9    0/1     ContainerCreating   0          54s
syndesis-server-1-56tnv        0/1     ContainerCreating   0          59s
syndesis-server-1-deploy       1/1     Running             0          67s
syndesis-ui-1-5zjtv            0/1     ContainerCreating   0          59s
syndesis-ui-1-deploy           1/1     Running             0          68s
$ oc describe --namespace syndesis pod syndesis-operator-1-spthx
Name:         syndesis-operator-1-spthx
Namespace:    syndesis
Priority:     0
Node:         crc-pkjt4-master-0/192.168.126.11
Start Time:   Thu, 01 Jul 2021 11:08:06 +0200
Labels:       deployment=syndesis-operator-1
              deploymentconfig=syndesis-operator
              name=syndesis-operator
              syndesis.io/app=syndesis
              syndesis.io/component=syndesis-operator
              syndesis.io/type=operator
Annotations:  k8s.v1.cni.cncf.io/network-status:
                [{
                    "name": "",
                    "interface": "eth0",
                    "ips": [
                        "10.217.0.60"
                    ],
                    "default": true,
                    "dns": {}
                }]
              k8s.v1.cni.cncf.io/networks-status:
                [{
                    "name": "",
                    "interface": "eth0",
                    "ips": [
                        "10.217.0.60"
                    ],
                    "default": true,
                    "dns": {}
                }]
              openshift.io/deployment-config.latest-version: 1
              openshift.io/deployment-config.name: syndesis-operator
              openshift.io/deployment.name: syndesis-operator-1
              openshift.io/scc: restricted
Status:       Running
IP:           10.217.0.60
IPs:
  IP:           10.217.0.60
Controlled By:  ReplicationController/syndesis-operator-1
Init Containers:
  postgres-version:
    Container ID:  cri-o://fc0a08e2117eda6d0fcd8edfe97656983d8a6b8a050a0b2e3d3d7fe006d070ca
    Image:         centos/postgresql-10-centos7
    Image ID:      docker.io/centos/postgresql-10-centos7@sha256:9e176b04e86d21560436b63c67b2e92f036879f5854e85598e2c08e5a7d3dc68
    Port:          <none>
    Host Port:     <none>
    Command:
      bash
      -c
      postgres -V > /data/postgresql.txt
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Thu, 01 Jul 2021 11:08:38 +0200
      Finished:     Thu, 01 Jul 2021 11:08:39 +0200
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /data from syndesis-operator-data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from syndesis-operator-token-wxkcr (ro)
Containers:
  syndesis-operator:
    Container ID:  cri-o://f985025b0a9e83c8ca72cd12c38eb5277848402c740ef86064958a789bda13bf
    Image:         syndesis/syndesis-operator@sha256:0f957a65dbd4b5e6fe94bfcd70c4710ee630e1cf2dbf29906a21035bf5ea42a1
    Image ID:      docker.io/syndesis/syndesis-operator@sha256:0f957a65dbd4b5e6fe94bfcd70c4710ee630e1cf2dbf29906a21035bf5ea42a1
    Port:          60000/TCP
    Host Port:     0/TCP
    Args:
      --zap-level=info
    State:          Running
      Started:      Thu, 01 Jul 2021 11:09:01 +0200
    Ready:          True
    Restart Count:  0
    Environment:
      WATCH_NAMESPACE:  syndesis (v1:metadata.namespace)
      POD_NAME:         syndesis-operator-1-spthx (v1:metadata.name)
      OPERATOR_NAME:    syndesis-operator
      DEV_SUPPORT:      false
    Mounts:
      /data from syndesis-operator-data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from syndesis-operator-token-wxkcr (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  syndesis-operator-data:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     
    SizeLimit:  <unset>
  syndesis-operator-token-wxkcr:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  syndesis-operator-token-wxkcr
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type    Reason          Age    From               Message
  ----    ------          ----   ----               -------
  Normal  Scheduled       2m50s  default-scheduler  Successfully assigned syndesis/syndesis-operator-1-spthx to crc-pkjt4-master-0
  Normal  AddedInterface  2m48s  multus             Add eth0 [10.217.0.60/23]
  Normal  Pulling         2m47s  kubelet            Pulling image "centos/postgresql-10-centos7"
  Normal  Pulled          2m18s  kubelet            Successfully pulled image "centos/postgresql-10-centos7" in 28.687897283s
  Normal  Created         2m18s  kubelet            Created container postgres-version
  Normal  Started         2m17s  kubelet            Started container postgres-version
  Normal  Pulling         2m17s  kubelet            Pulling image "syndesis/syndesis-operator@sha256:0f957a65dbd4b5e6fe94bfcd70c4710ee630e1cf2dbf29906a21035bf5ea42a1"
  Normal  Pulled          116s   kubelet            Successfully pulled image "syndesis/syndesis-operator@sha256:0f957a65dbd4b5e6fe94bfcd70c4710ee630e1cf2dbf29906a21035bf5ea42a1" in 20.813405555s
  Normal  Created         115s   kubelet            Created container syndesis-operator
  Normal  Started         115s   kubelet            Started container syndesis-operator
```

The issue you're facing could be caused by something in your environment. Hard to say what.

**Comment by mkralik3** [comment]
Hi. This 
```
error: no matches for kind "Syndesis" in version "syndesis.io/v1beta1"
```
was happened to me when I installed newer version of 1.12.x (after some CRD changes) and after that tried to install some older version of syndesis (e.g. 1.11.x).  There was an issue in 1.11 that during the installation  the CRD is not updated when it exists (already fixed in 1.12). So the workaround is remove syndesis CRD from cluster before you install 1.11 but be sure that you have not any other syndesis in the cluster because that would delete them.
